jetstream/cli.py

```diff
@@ -533,17 +534,40 @@ def validate_config(path):
     for file in config_files:
         click.echo(f"Validate {file}", err=False)
 
-        custom_spec = AnalysisSpec.from_dict(toml.load(file))
+        if Path(file).parent.name == OUTCOMES_DIR:
+            # file is an outcome snippet
+            outcomes_spec = OutcomeSpec.from_dict(toml.load(file))
+            dummy_experiment = Experiment(
+                experimenter_slug="dummy-experiment",
+                normandy_slug="dummy_experiment",
+                type="v6",
+                status="Live",
+                branches=[],
+                probe_sets=[],
+                end_date=None,
+                reference_branch="control",
+                is_high_population=False,
+                start_date=datetime.now(),
+                proposed_enrollment=14,
+                app_id="firefox-desktop",
+                app_name="firefox_desktop",
+            )
+            spec = AnalysisSpec.default_for_experiment(dummy_experiment)
+            spec.merge(outcomes_spec)
+            conf = spec.resolve(dummy_experiment)
+        else:
+            # validate experiment configuration file
+            custom_spec = AnalysisSpec.from_dict(toml.load(file))
```

Comment:
The idea here is to reuse `validate_config` and check if the modified file is in a `outcomes/` directory. To validate outcome snippets, a dummy experiment is initialized and the outcome snippet is merged into it's config. Then it can be dry run.

Another option here for validation is to retrieve all experiments from Experimenter that have this outcome specified and perform the dry runs with those. However, initially when a new outcome is added no experiments will have this outcome set, so this only works when modifying an existing outcome snippet.